# While Loops

In [1]:
i = 1
while i <= 5:
    print("India")
    i += 1

India
India
India
India
India


In [2]:
i = 1
while i <= 5:
    print(i)
    i += 1

1
2
3
4
5


In [3]:
i = 1
while i <= 5:
    print(i)
    i += 1
print("Finish")

1
2
3
4
5
Finish


In [4]:
i = 3
while i >= 0:
    print(i)
    i -= 1

3
2
1
0


# Break Statement

In [5]:
i = 1
while True:
    if i%2 == 0:
        break
    print(i)
    i += 2

1
3
5
7
9
11
13
15
17
19
21
23
25
27
29
31
33
35
37
39
41
43
45
47
49
51
53
55
57
59
61
63
65
67
69
71
73
75
77
79
81
83
85
87
89
91
93
95
97
99
101
103
105
107
109
111
113
115
117
119
121
123
125
127
129
131
133
135
137
139
141
143
145
147
149
151
153
155
157
159
161
163
165
167
169
171
173
175
177
179
181
183
185
187
189
191
193
195
197
199
201
203
205
207
209
211
213
215
217
219
221
223
225
227
229
231
233
235
237
239
241
243
245
247
249
251
253
255
257
259
261
263
265
267
269
271
273
275
277
279
281
283
285
287
289
291
293
295
297
299
301
303
305
307
309
311
313
315
317
319
321
323
325
327
329
331
333
335
337
339
341
343
345
347
349
351
353
355
357
359
361
363
365
367
369
371
373
375
377
379
381
383
385
387
389
391
393
395
397
399
401
403
405
407
409
411
413
415
417
419
421
423
425
427
429
431
433
435
437
439
441
443
445
447
449
451
453
455
457
459
461
463
465
467
469
471
473
475
477
479
481
483
485
487
489
491
493
495
497
499
501
503
505
507
509
511
513
515
517
519
521
523
525
527

5683
5685
5687
5689
5691
5693
5695
5697
5699
5701
5703
5705
5707
5709
5711
5713
5715
5717
5719
5721
5723
5725
5727
5729
5731
5733
5735
5737
5739
5741
5743
5745
5747
5749
5751
5753
5755
5757
5759
5761
5763
5765
5767
5769
5771
5773
5775
5777
5779
5781
5783
5785
5787
5789
5791
5793
5795
5797
5799
5801
5803
5805
5807
5809
5811
5813
5815
5817
5819
5821
5823
5825
5827
5829
5831
5833
5835
5837
5839
5841
5843
5845
5847
5849
5851
5853
5855
5857
5859
5861
5863
5865
5867
5869
5871
5873
5875
5877
5879
5881
5883
5885
5887
5889
5891
5893
5895
5897
5899
5901
5903
5905
5907
5909
5911
5913
5915
5917
5919
5921
5923
5925
5927
5929
5931
5933
5935
5937
5939
5941
5943
5945
5947
5949
5951
5953
5955
5957
5959
5961
5963
5965
5967
5969
5971
5973
5975
5977
5979
5981
5983
5985
5987
5989
5991
5993
5995
5997
5999
6001
6003
6005
6007
6009
6011
6013
6015
6017
6019
6021
6023
6025
6027
6029
6031
6033
6035
6037
6039
6041
6043
6045
6047
6049
6051
6053
6055
6057
6059
6061
6063
6065
6067
6069
6071
6073
6075
6077
6079
6081


10319
10321
10323
10325
10327
10329
10331
10333
10335
10337
10339
10341
10343
10345
10347
10349
10351
10353
10355
10357
10359
10361
10363
10365
10367
10369
10371
10373
10375
10377
10379
10381
10383
10385
10387
10389
10391
10393
10395
10397
10399
10401
10403
10405
10407
10409
10411
10413
10415
10417
10419
10421
10423
10425
10427
10429
10431
10433
10435
10437
10439
10441
10443
10445
10447
10449
10451
10453
10455
10457
10459
10461
10463
10465
10467
10469
10471
10473
10475
10477
10479
10481
10483
10485
10487
10489
10491
10493
10495
10497
10499
10501
10503
10505
10507
10509
10511
10513
10515
10517
10519
10521
10523
10525
10527
10529
10531
10533
10535
10537
10539
10541
10543
10545
10547
10549
10551
10553
10555
10557
10559
10561
10563
10565
10567
10569
10571
10573
10575
10577
10579
10581
10583
10585
10587
10589
10591
10593
10595
10597
10599
10601
10603
10605
10607
10609
10611
10613
10615
10617
10619
10621
10623
10625
10627
10629
10631
10633
10635
10637
10639
10641
10643
10645
10647
10649
1065

13877
13879
13881
13883
13885
13887
13889
13891
13893
13895
13897
13899
13901
13903
13905
13907
13909
13911
13913
13915
13917
13919
13921
13923
13925
13927
13929
13931
13933
13935
13937
13939
13941
13943
13945
13947
13949
13951
13953
13955
13957
13959
13961
13963
13965
13967
13969
13971
13973
13975
13977
13979
13981
13983
13985
13987
13989
13991
13993
13995
13997
13999
14001
14003
14005
14007
14009
14011
14013
14015
14017
14019
14021
14023
14025
14027
14029
14031
14033
14035
14037
14039
14041
14043
14045
14047
14049
14051
14053
14055
14057
14059
14061
14063
14065
14067
14069
14071
14073
14075
14077
14079
14081
14083
14085
14087
14089
14091
14093
14095
14097
14099
14101
14103
14105
14107
14109
14111
14113
14115
14117
14119
14121
14123
14125
14127
14129
14131
14133
14135
14137
14139
14141
14143
14145
14147
14149
14151
14153
14155
14157
14159
14161
14163
14165
14167
14169
14171
14173
14175
14177
14179
14181
14183
14185
14187
14189
14191
14193
14195
14197
14199
14201
14203
14205
14207
1420

18119
18121
18123
18125
18127
18129
18131
18133
18135
18137
18139
18141
18143
18145
18147
18149
18151
18153
18155
18157
18159
18161
18163
18165
18167
18169
18171
18173
18175
18177
18179
18181
18183
18185
18187
18189
18191
18193
18195
18197
18199
18201
18203
18205
18207
18209
18211
18213
18215
18217
18219
18221
18223
18225
18227
18229
18231
18233
18235
18237
18239
18241
18243
18245
18247
18249
18251
18253
18255
18257
18259
18261
18263
18265
18267
18269
18271
18273
18275
18277
18279
18281
18283
18285
18287
18289
18291
18293
18295
18297
18299
18301
18303
18305
18307
18309
18311
18313
18315
18317
18319
18321
18323
18325
18327
18329
18331
18333
18335
18337
18339
18341
18343
18345
18347
18349
18351
18353
18355
18357
18359
18361
18363
18365
18367
18369
18371
18373
18375
18377
18379
18381
18383
18385
18387
18389
18391
18393
18395
18397
18399
18401
18403
18405
18407
18409
18411
18413
18415
18417
18419
18421
18423
18425
18427
18429
18431
18433
18435
18437
18439
18441
18443
18445
18447
18449
1845

22873
22875
22877
22879
22881
22883
22885
22887
22889
22891
22893
22895
22897
22899
22901
22903
22905
22907
22909
22911
22913
22915
22917
22919
22921
22923
22925
22927
22929
22931
22933
22935
22937
22939
22941
22943
22945
22947
22949
22951
22953
22955
22957
22959
22961
22963
22965
22967
22969
22971
22973
22975
22977
22979
22981
22983
22985
22987
22989
22991
22993
22995
22997
22999
23001
23003
23005
23007
23009
23011
23013
23015
23017
23019
23021
23023
23025
23027
23029
23031
23033
23035
23037
23039
23041
23043
23045
23047
23049
23051
23053
23055
23057
23059
23061
23063
23065
23067
23069
23071
23073
23075
23077
23079
23081
23083
23085
23087
23089
23091
23093
23095
23097
23099
23101
23103
23105
23107
23109
23111
23113
23115
23117
23119
23121
23123
23125
23127
23129
23131
23133
23135
23137
23139
23141
23143
23145
23147
23149
23151
23153
23155
23157
23159
23161
23163
23165
23167
23169
23171
23173
23175
23177
23179
23181
23183
23185
23187
23189
23191
23193
23195
23197
23199
23201
23203
2320

25873
25875
25877
25879
25881
25883
25885
25887
25889
25891
25893
25895
25897
25899
25901
25903
25905
25907
25909
25911
25913
25915
25917
25919
25921
25923
25925
25927
25929
25931
25933
25935
25937
25939
25941
25943
25945
25947
25949
25951
25953
25955
25957
25959
25961
25963
25965
25967
25969
25971
25973
25975
25977
25979
25981
25983
25985
25987
25989
25991
25993
25995
25997
25999
26001
26003
26005
26007
26009
26011
26013
26015
26017
26019
26021
26023
26025
26027
26029
26031
26033
26035
26037
26039
26041
26043
26045
26047
26049
26051
26053
26055
26057
26059
26061
26063
26065
26067
26069
26071
26073
26075
26077
26079
26081
26083
26085
26087
26089
26091
26093
26095
26097
26099
26101
26103
26105
26107
26109
26111
26113
26115
26117
26119
26121
26123
26125
26127
26129
26131
26133
26135
26137
26139
26141
26143
26145
26147
26149
26151
26153
26155
26157
26159
26161
26163
26165
26167
26169
26171
26173
26175
26177
26179
26181
26183
26185
26187
26189
26191
26193
26195
26197
26199
26201
26203
2620

28871
28873
28875
28877
28879
28881
28883
28885
28887
28889
28891
28893
28895
28897
28899
28901
28903
28905
28907
28909
28911
28913
28915
28917
28919
28921
28923
28925
28927
28929
28931
28933
28935
28937
28939
28941
28943
28945
28947
28949
28951
28953
28955
28957
28959
28961
28963
28965
28967
28969
28971
28973
28975
28977
28979
28981
28983
28985
28987
28989
28991
28993
28995
28997
28999
29001
29003
29005
29007
29009
29011
29013
29015
29017
29019
29021
29023
29025
29027
29029
29031
29033
29035
29037
29039
29041
29043
29045
29047
29049
29051
29053
29055
29057
29059
29061
29063
29065
29067
29069
29071
29073
29075
29077
29079
29081
29083
29085
29087
29089
29091
29093
29095
29097
29099
29101
29103
29105
29107
29109
29111
29113
29115
29117
29119
29121
29123
29125
29127
29129
29131
29133
29135
29137
29139
29141
29143
29145
29147
29149
29151
29153
29155
29157
29159
29161
29163
29165
29167
29169
29171
29173
29175
29177
29179
29181
29183
29185
29187
29189
29191
29193
29195
29197
29199
29201
2920

33869
33871
33873
33875
33877
33879
33881
33883
33885
33887
33889
33891
33893
33895
33897
33899
33901
33903
33905
33907
33909
33911
33913
33915
33917
33919
33921
33923
33925
33927
33929
33931
33933
33935
33937
33939
33941
33943
33945
33947
33949
33951
33953
33955
33957
33959
33961
33963
33965
33967
33969
33971
33973
33975
33977
33979
33981
33983
33985
33987
33989
33991
33993
33995
33997
33999
34001
34003
34005
34007
34009
34011
34013
34015
34017
34019
34021
34023
34025
34027
34029
34031
34033
34035
34037
34039
34041
34043
34045
34047
34049
34051
34053
34055
34057
34059
34061
34063
34065
34067
34069
34071
34073
34075
34077
34079
34081
34083
34085
34087
34089
34091
34093
34095
34097
34099
34101
34103
34105
34107
34109
34111
34113
34115
34117
34119
34121
34123
34125
34127
34129
34131
34133
34135
34137
34139
34141
34143
34145
34147
34149
34151
34153
34155
34157
34159
34161
34163
34165
34167
34169
34171
34173
34175
34177
34179
34181
34183
34185
34187
34189
34191
34193
34195
34197
34199
3420

38265
38267
38269
38271
38273
38275
38277
38279
38281
38283
38285
38287
38289
38291
38293
38295
38297
38299
38301
38303
38305
38307
38309
38311
38313
38315
38317
38319
38321
38323
38325
38327
38329
38331
38333
38335
38337
38339
38341
38343
38345
38347
38349
38351
38353
38355
38357
38359
38361
38363
38365
38367
38369
38371
38373
38375
38377
38379
38381
38383
38385
38387
38389
38391
38393
38395
38397
38399
38401
38403
38405
38407
38409
38411
38413
38415
38417
38419
38421
38423
38425
38427
38429
38431
38433
38435
38437
38439
38441
38443
38445
38447
38449
38451
38453
38455
38457
38459
38461
38463
38465
38467
38469
38471
38473
38475
38477
38479
38481
38483
38485
38487
38489
38491
38493
38495
38497
38499
38501
38503
38505
38507
38509
38511
38513
38515
38517
38519
38521
38523
38525
38527
38529
38531
38533
38535
38537
38539
38541
38543
38545
38547
38549
38551
38553
38555
38557
38559
38561
38563
38565
38567
38569
38571
38573
38575
38577
38579
38581
38583
38585
38587
38589
38591
38593
38595
3859

42865
42867
42869
42871
42873
42875
42877
42879
42881
42883
42885
42887
42889
42891
42893
42895
42897
42899
42901
42903
42905
42907
42909
42911
42913
42915
42917
42919
42921
42923
42925
42927
42929
42931
42933
42935
42937
42939
42941
42943
42945
42947
42949
42951
42953
42955
42957
42959
42961
42963
42965
42967
42969
42971
42973
42975
42977
42979
42981
42983
42985
42987
42989
42991
42993
42995
42997
42999
43001
43003
43005
43007
43009
43011
43013
43015
43017
43019
43021
43023
43025
43027
43029
43031
43033
43035
43037
43039
43041
43043
43045
43047
43049
43051
43053
43055
43057
43059
43061
43063
43065
43067
43069
43071
43073
43075
43077
43079
43081
43083
43085
43087
43089
43091
43093
43095
43097
43099
43101
43103
43105
43107
43109
43111
43113
43115
43117
43119
43121
43123
43125
43127
43129
43131
43133
43135
43137
43139
43141
43143
43145
43147
43149
43151
43153
43155
43157
43159
43161
43163
43165
43167
43169
43171
43173
43175
43177
43179
43181
43183
43185
43187
43189
43191
43193
43195
4319

46371
46373
46375
46377
46379
46381
46383
46385
46387
46389
46391
46393
46395
46397
46399
46401
46403
46405
46407
46409
46411
46413
46415
46417
46419
46421
46423
46425
46427
46429
46431
46433
46435
46437
46439
46441
46443
46445
46447
46449
46451
46453
46455
46457
46459
46461
46463
46465
46467
46469
46471
46473
46475
46477
46479
46481
46483
46485
46487
46489
46491
46493
46495
46497
46499
46501
46503
46505
46507
46509
46511
46513
46515
46517
46519
46521
46523
46525
46527
46529
46531
46533
46535
46537
46539
46541
46543
46545
46547
46549
46551
46553
46555
46557
46559
46561
46563
46565
46567
46569
46571
46573
46575
46577
46579
46581
46583
46585
46587
46589
46591
46593
46595
46597
46599
46601
46603
46605
46607
46609
46611
46613
46615
46617
46619
46621
46623
46625
46627
46629
46631
46633
46635
46637
46639
46641
46643
46645
46647
46649
46651
46653
46655
46657
46659
46661
46663
46665
46667
46669
46671
46673
46675
46677
46679
46681
46683
46685
46687
46689
46691
46693
46695
46697
46699
46701
4670

51803
51805
51807
51809
51811
51813
51815
51817
51819
51821
51823
51825
51827
51829
51831
51833
51835
51837
51839
51841
51843
51845
51847
51849
51851
51853
51855
51857
51859
51861
51863
51865
51867
51869
51871
51873
51875
51877
51879
51881
51883
51885
51887
51889
51891
51893
51895
51897
51899
51901
51903
51905
51907
51909
51911
51913
51915
51917
51919
51921
51923
51925
51927
51929
51931
51933
51935
51937
51939
51941
51943
51945
51947
51949
51951
51953
51955
51957
51959
51961
51963
51965
51967
51969
51971
51973
51975
51977
51979
51981
51983
51985
51987
51989
51991
51993
51995
51997
51999
52001
52003
52005
52007
52009
52011
52013
52015
52017
52019
52021
52023
52025
52027
52029
52031
52033
52035
52037
52039
52041
52043
52045
52047
52049
52051
52053
52055
52057
52059
52061
52063
52065
52067
52069
52071
52073
52075
52077
52079
52081
52083
52085
52087
52089
52091
52093
52095
52097
52099
52101
52103
52105
52107
52109
52111
52113
52115
52117
52119
52121
52123
52125
52127
52129
52131
52133
5213

55859
55861
55863
55865
55867
55869
55871
55873
55875
55877
55879
55881
55883
55885
55887
55889
55891
55893
55895
55897
55899
55901
55903
55905
55907
55909
55911
55913
55915
55917
55919
55921
55923
55925
55927
55929
55931
55933
55935
55937
55939
55941
55943
55945
55947
55949
55951
55953
55955
55957
55959
55961
55963
55965
55967
55969
55971
55973
55975
55977
55979
55981
55983
55985
55987
55989
55991
55993
55995
55997
55999
56001
56003
56005
56007
56009
56011
56013
56015
56017
56019
56021
56023
56025
56027
56029
56031
56033
56035
56037
56039
56041
56043
56045
56047
56049
56051
56053
56055
56057
56059
56061
56063
56065
56067
56069
56071
56073
56075
56077
56079
56081
56083
56085
56087
56089
56091
56093
56095
56097
56099
56101
56103
56105
56107
56109
56111
56113
56115
56117
56119
56121
56123
56125
56127
56129
56131
56133
56135
56137
56139
56141
56143
56145
56147
56149
56151
56153
56155
56157
56159
56161
56163
56165
56167
56169
56171
56173
56175
56177
56179
56181
56183
56185
56187
56189
5619

60783
60785
60787
60789
60791
60793
60795
60797
60799
60801
60803
60805
60807
60809
60811
60813
60815
60817
60819
60821
60823
60825
60827
60829
60831
60833
60835
60837
60839
60841
60843
60845
60847
60849
60851
60853
60855
60857
60859
60861
60863
60865
60867
60869
60871
60873
60875
60877
60879
60881
60883
60885
60887
60889
60891
60893
60895
60897
60899
60901
60903
60905
60907
60909
60911
60913
60915
60917
60919
60921
60923
60925
60927
60929
60931
60933
60935
60937
60939
60941
60943
60945
60947
60949
60951
60953
60955
60957
60959
60961
60963
60965
60967
60969
60971
60973
60975
60977
60979
60981
60983
60985
60987
60989
60991
60993
60995
60997
60999
61001
61003
61005
61007
61009
61011
61013
61015
61017
61019
61021
61023
61025
61027
61029
61031
61033
61035
61037
61039
61041
61043
61045
61047
61049
61051
61053
61055
61057
61059
61061
61063
61065
61067
61069
61071
61073
61075
61077
61079
61081
61083
61085
61087
61089
61091
61093
61095
61097
61099
61101
61103
61105
61107
61109
61111
61113
6111

65423
65425
65427
65429
65431
65433
65435
65437
65439
65441
65443
65445
65447
65449
65451
65453
65455
65457
65459
65461
65463
65465
65467
65469
65471
65473
65475
65477
65479
65481
65483
65485
65487
65489
65491
65493
65495
65497
65499
65501
65503
65505
65507
65509
65511
65513
65515
65517
65519
65521
65523
65525
65527
65529
65531
65533
65535
65537
65539
65541
65543
65545
65547
65549
65551
65553
65555
65557
65559
65561
65563
65565
65567
65569
65571
65573
65575
65577
65579
65581
65583
65585
65587
65589
65591
65593
65595
65597
65599
65601
65603
65605
65607
65609
65611
65613
65615
65617
65619
65621
65623
65625
65627
65629
65631
65633
65635
65637
65639
65641
65643
65645
65647
65649
65651
65653
65655
65657
65659
65661
65663
65665
65667
65669
65671
65673
65675
65677
65679
65681
65683
65685
65687
65689
65691
65693
65695
65697
65699
65701
65703
65705
65707
65709
65711
65713
65715
65717
65719
65721
65723
65725
65727
65729
65731
65733
65735
65737
65739
65741
65743
65745
65747
65749
65751
65753
6575

68599
68601
68603
68605
68607
68609
68611
68613
68615
68617
68619
68621
68623
68625
68627
68629
68631
68633
68635
68637
68639
68641
68643
68645
68647
68649
68651
68653
68655
68657
68659
68661
68663
68665
68667
68669
68671
68673
68675
68677
68679
68681
68683
68685
68687
68689
68691
68693
68695
68697
68699
68701
68703
68705
68707
68709
68711
68713
68715
68717
68719
68721
68723
68725
68727
68729
68731
68733
68735
68737
68739
68741
68743
68745
68747
68749
68751
68753
68755
68757
68759
68761
68763
68765
68767
68769
68771
68773
68775
68777
68779
68781
68783
68785
68787
68789
68791
68793
68795
68797
68799
68801
68803
68805
68807
68809
68811
68813
68815
68817
68819
68821
68823
68825
68827
68829
68831
68833
68835
68837
68839
68841
68843
68845
68847
68849
68851
68853
68855
68857
68859
68861
68863
68865
68867
68869
68871
68873
68875
68877
68879
68881
68883
68885
68887
68889
68891
68893
68895
68897
68899
68901
68903
68905
68907
68909
68911
68913
68915
68917
68919
68921
68923
68925
68927
68929
6893

71853
71855
71857
71859
71861
71863
71865
71867
71869
71871
71873
71875
71877
71879
71881
71883
71885
71887
71889
71891
71893
71895
71897
71899
71901
71903
71905
71907
71909
71911
71913
71915
71917
71919
71921
71923
71925
71927
71929
71931
71933
71935
71937
71939
71941
71943
71945
71947
71949
71951
71953
71955
71957
71959
71961
71963
71965
71967
71969
71971
71973
71975
71977
71979
71981
71983
71985
71987
71989
71991
71993
71995
71997
71999
72001
72003
72005
72007
72009
72011
72013
72015
72017
72019
72021
72023
72025
72027
72029
72031
72033
72035
72037
72039
72041
72043
72045
72047
72049
72051
72053
72055
72057
72059
72061
72063
72065
72067
72069
72071
72073
72075
72077
72079
72081
72083
72085
72087
72089
72091
72093
72095
72097
72099
72101
72103
72105
72107
72109
72111
72113
72115
72117
72119
72121
72123
72125
72127
72129
72131
72133
72135
72137
72139
72141
72143
72145
72147
72149
72151
72153
72155
72157
72159
72161
72163
72165
72167
72169
72171
72173
72175
72177
72179
72181
72183
7218

75851
75853
75855
75857
75859
75861
75863
75865
75867
75869
75871
75873
75875
75877
75879
75881
75883
75885
75887
75889
75891
75893
75895
75897
75899
75901
75903
75905
75907
75909
75911
75913
75915
75917
75919
75921
75923
75925
75927
75929
75931
75933
75935
75937
75939
75941
75943
75945
75947
75949
75951
75953
75955
75957
75959
75961
75963
75965
75967
75969
75971
75973
75975
75977
75979
75981
75983
75985
75987
75989
75991
75993
75995
75997
75999
76001
76003
76005
76007
76009
76011
76013
76015
76017
76019
76021
76023
76025
76027
76029
76031
76033
76035
76037
76039
76041
76043
76045
76047
76049
76051
76053
76055
76057
76059
76061
76063
76065
76067
76069
76071
76073
76075
76077
76079
76081
76083
76085
76087
76089
76091
76093
76095
76097
76099
76101
76103
76105
76107
76109
76111
76113
76115
76117
76119
76121
76123
76125
76127
76129
76131
76133
76135
76137
76139
76141
76143
76145
76147
76149
76151
76153
76155
76157
76159
76161
76163
76165
76167
76169
76171
76173
76175
76177
76179
76181
7618

79277
79279
79281
79283
79285
79287
79289
79291
79293
79295
79297
79299
79301
79303
79305
79307
79309
79311
79313
79315
79317
79319
79321
79323
79325
79327
79329
79331
79333
79335
79337
79339
79341
79343
79345
79347
79349
79351
79353
79355
79357
79359
79361
79363
79365
79367
79369
79371
79373
79375
79377
79379
79381
79383
79385
79387
79389
79391
79393
79395
79397
79399
79401
79403
79405
79407
79409
79411
79413
79415
79417
79419
79421
79423
79425
79427
79429
79431
79433
79435
79437
79439
79441
79443
79445
79447
79449
79451
79453
79455
79457
79459
79461
79463
79465
79467
79469
79471
79473
79475
79477
79479
79481
79483
79485
79487
79489
79491
79493
79495
79497
79499
79501
79503
79505
79507
79509
79511
79513
79515
79517
79519
79521
79523
79525
79527
79529
79531
79533
79535
79537
79539
79541
79543
79545
79547
79549
79551
79553
79555
79557
79559
79561
79563
79565
79567
79569
79571
79573
79575
79577
79579
79581
79583
79585
79587
79589
79591
79593
79595
79597
79599
79601
79603
79605
79607
7960

KeyboardInterrupt: 

In [6]:
i = 2
while True:
    if i%3 == 0:
        break
    print(i)
    i += 2

2
4


In [7]:
i = 2
while True:
    if i%3 == 0:
        break
    print(i)
    i += 2
print("HI")

2
4
HI


# Continue Statement

In [8]:
i = 2
while True:
    if i%3 == 0:
        continue
    print(i)
    i += 2

2
4


KeyboardInterrupt: 

In [11]:
i = 1
while i <= 10:
    i += 1
    if i%2 == 0:
        continue
    else:
        print(i)

3
5
7
9
11


# Print Statement without line change

In [19]:
print("Hello", end="")
print("Bye")

HelloBye


# Print following Patterns

1. 1  2  3  4  5  6  7  8  9  10
2. 1  4  9  16  25  36  49  64  81  100

In [21]:
i = 1
while i <= 10:
    print(i, end="  ")
    i += 1

1  2  3  4  5  6  7  8  9  10  

In [22]:
i = 1
while i <= 10:
    print(i*i, end="  ")
    i += 1

1  4  9  16  25  36  49  64  81  100  

# Questions

**1. WAP to enter 10 integers from user and print their average value on the screen.**

In [24]:
i = 1
s = 0
while i <= 10:
    a = int(input("Enter a number: "))
    s = s + a
    i += 1
print("Sum = ", s)
print("Avg = ", (s/10))

Enter a number: 1
Enter a number: 2
Enter a number: 3
Enter a number: 4
Enter a number: 5
Enter a number: 6
Enter a number: 7
Enter a number: 8
Enter a number: 9
Enter a number: 10
Sum =  55
Avg =  5.5


**2. Print multiplication table of 8 using loop.**

In [25]:
i = 1
while i <= 10:
    print(i*8)
    i += 1

8
16
24
32
40
48
56
64
72
80


**3. WAP to enter a number from user and print reverse of it.**

In [27]:
x = int(input("Enter a number : "))  #132    --  231
while x > 0:
    r = x%10
    print(r, end="")
    x = x//10

Enter a number : 153
351

In [3]:
x = int(input("Enter a number : "))  #132    --  231
s = 0
while x > 0:
    r = x%10
    s = s*10 + r
    x = x//10
print(s)

Enter a number : 153
351


**4. WAP to enter a number from user and print addition of its individual digits.**

In [29]:
x = int(input("Enter a number : "))  #132    --  231
s = 0
while x > 0:
    r = x%10
    s = s + r
    x = x//10
print("Addition = ", s)

Enter a number : 123
Addition =  6


**5. WAP to enter a number from user and check it is a palindrome or not.**

121 <br>
12321

In [5]:
x = int(input("Enter a number : "))
s = 0
temp = x
while x > 0:
    r = x%10
    s = s*10 + r
    x = x//10
if s == temp:
    print("Palindrome")
else:
    print("Not Palindrome")

Enter a number : 121
Palindrome


**6. WAP to enter a number from user and check it is an armstrong number or not.**

153 = cube(1) +cube(5) +cube(3)

**7. WAP to enter a number from user and print individual digits on separate line.**

153<br>
1<br>
5<br>
3<br>

In [7]:
x = int(input("Enter a number : "))
s = 0
while x > 0:
    r = x%10
    s = s*10 + r
    x = x//10

while s > 0:
    r = s%10
    print(r)
    s = s//10

Enter a number : 153
1
5
3


**8. WAP to enter a number from user and print individual digits in character formate.**

122 - "One Two Two"

In [9]:
x = int(input("Enter a number : "))
s = 0
while x > 0:
    r = x%10
    s = s*10 + r
    x = x//10

while s > 0:
    r = s%10
    if r == 0:
        print("Zero")
    elif r ==1:
        print("One")
    elif r ==2:
        print("Two")
    elif r ==3:
        print("Three")
    elif r ==4:
        print("Four")
    elif r ==5:
        print("Five")
    elif r ==6:
        print("Six")
    elif r ==7:
        print("Seven")
    elif r ==8:
        print("Eight")
    elif r ==9:
        print("Nine")
    s = s//10

Enter a number : 122
One
Two
Two


**9. WAP for factorial of a number.**

In [14]:
n = int(input("Enter a number : "))
f = 1
while n>1:
    f = f * n    #f = 1, 5, 20, 60, 120
    n -= 1       #n = 5, 4, 3, 2, 1
print("Factorial = ", f)

Enter a number : 0
Factorial =  1


**10. WAP to print Fabonacci Series of numbers.**<br>
0 1 1 2 3 5 8<br>
a b c<br>
  a b c

In [15]:
10 -- 3  #10 - (-3)

13

# Do It Yourself
Print following patterns
1. 1/1   2/4   3/9   4/16   5/25   6/36   7/49   8/64   9/81   10/100
2. 1/1 + 2/4 + 3/9 + 4/16 + 5/25 + 6/36 + 7/49 + 8/64 + 9/81 + 10/100

# Competitive Programming Questions

**1. Find the sum of all the multiples of 3 or 5 below 1000**

Hint : all numbers below 10 will be 3, 5, 6 and 9. Sum of these multiples is 23.
<br> <br> <br> 
**2. Fabonacci starting with 1 and 2, the first 10 terms will be : <br> 1,2,3,5,8,12,21,34,55,89. <br> Find the sum of even valued term within 4 million.**
<br> <br> 

**3. 2520 is the smallest number that can be divisible by each number from 1 to 10. What is the smallest positive number that is evenly divisible by all the numbers from 1 to 20?**

# Loop Else

In [16]:
i = 0
while i < 5:
    print(i)
    i+=1
    if i==3:
        break
else:
    print(0)

0
1
2


In [17]:
i = 0
while i < 3:
    print(i)
    i+=1
else:
    print(0)

0
1
2
0


In [18]:
i = 0
while True:
    i += 1
    if i==2:
        continue
    if i==5:
        break
    print(i)
else:
    print("Finished")

1
3
4


# Nested While Loop

In [ ]:
11   12   13
21   22   23
31   32   33

In [21]:
i = 1
while i<=3:
    j = 1
    while j<=3:
        print(i,j, end="\t")
        j += 1
    print()
    i += 1


1 1	1 2	1 3	
2 1	2 2	2 3	
3 1	3 2	3 3	


# Printing Patterns

In [ ]:
*
**
***
****
*****

In [ ]:
1
12
123
1234
12345

In [ ]:
1
22
333
4444
55555

In [ ]:
5
44
333
2222
11111

In [ ]:
5
54
543
5432
54321

In [ ]:
55555
4444
333
22
1

In [ ]:
54321
5432
543
54
5

In [ ]:
11111
2222
333
44
5

In [ ]:
12345
1234
123
12
1

In [ ]:
1
23
456
78910

In [ ]:
    *
   **
  ***
 ****
*****

In [ ]:
    *
   ***
  *****
 *******
*********